File: greedy-algorithm/1이-될-때까지/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int result = 0;

        while (true) {
            int target = (n / k) * k;
            result += (n - target);
            n = target;

            if (n < k) {
                break;
            }

            n /= k;
            result += 1;
        }

        result += (n - 1);
        System.out.println(result);

        sc.close();
    }

}

File: greedy-algorithm/1이-될-때까지/main.cpp
#include <bits/stdc++.h>

using namespace std;

int n, k;
int result;

int main()
{
    cin >> n >> k;

    while (true)
    {
        int target = (n / k) * k;
        result += (n - target);
        n = target;

        if (n < k)
        {
            break;
        }

        n /= k;
        result += 1;
    }

    result += (n - 1);
    cout << result << '\n';
}
